src/impl.cpp

```diff
@@ -0,0 +1,14 @@
+#include "impl.h"
+
+#define INT32_BIT_LENGTH 32
```

Comment:
Also, I think it should be inside the scope of IntToBinaryRepresentation function